ext/io/wait/wait.c

```diff
@@ -393,9 +393,15 @@ Init_wait(void)
 
     rb_define_method(rb_cIO, "wait", io_wait, -1);
 
-    rb_define_method(rb_cIO, "wait_readable", io_wait_readable, -1);
-    rb_define_method(rb_cIO, "wait_writable", io_wait_writable, -1);
+    if (!RTEST(rb_funcall(rb_cIO, rb_intern("method_defined?"), 1, ID2SYM(rb_intern("wait_readable"))))) {
+        rb_define_method(rb_cIO, "wait_readable", io_wait_readable, -1);
+    }
+    if (!RTEST(rb_funcall(rb_cIO, rb_intern("method_defined?"), 1, ID2SYM(rb_intern("wait_writable"))))) {
+        rb_define_method(rb_cIO, "wait_writable", io_wait_writable, -1);
+    }
```

Comment:
cc @eregon

Reply:
Yes, that's safer I think and avoids hardcoding any version.